app/src/main/java/com/maasbodev/tasky/MainActivity.kt

```diff
@@ -4,42 +4,44 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
-import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.res.stringResource
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import com.maasbodev.tasky.ui.theme.TaskyTheme
 
+@OptIn(ExperimentalMaterial3Api::class)
 class MainActivity : ComponentActivity() {
 	override fun onCreate(savedInstanceState: Bundle?) {
 		super.onCreate(savedInstanceState)
 		enableEdgeToEdge()
 		setContent {
 			TaskyTheme {
-				Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
-					Greeting(
-						name = "Android", modifier = Modifier.padding(innerPadding)
-					)
+				val navController = rememberNavController()
+				val title = remember { mutableStateOf("") }
+				Scaffold(topBar = {
+					CenterAlignedTopAppBar(title = { Text(text = title.value) })
+				}) {
+					NavHost(
+						modifier = Modifier.padding(it),
+						navController = navController,
+						startDestination = Register
+					) {
+						composable<Register> {
+							title.value = stringResource(R.string.create_your_account)
```

Comment:
I'm not really sure if you intended to test something here or if this code should stay like that, but it does not make a lot of sense for two reasons:
1. The `Create your account` text on the register screen is static and never changes. State is used for values that change over time, so in this case you could assign that string resource right away to the text composable in the register screen.
2. You create the state outside of the screen (above the scaffold) which will make it live longer than the screen. I don't think this is desired behavior

Reply:
The state is in the MainActivity because is who has the TopAppBar that is using it as Title of each screen and every screen will change it.

Reply:
Okay, but every top bar in the app is quite different (especially compared with the login topbar). Could it make more sense to handle this on a screen level and not at root? You could still create reusable topbar composables of course that allow you to customize it.